langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModelIT.java

```diff
@@ -456,28 +491,186 @@ void should_use_tools_with_AiService_2() {
+    void should_work_with_interspersed_function_execution_results() {
+        // given
+        VertexAiGeminiChatModel model = VertexAiGeminiChatModel.builder()
+            .project("genai-java-demos")
+            .location("us-central1")
+            .modelName("gemini-1.5-flash-001")
+            .temperature(0.0f)
+            .topK(1)
+            .logRequests(true)
+            .logResponses(true)
+            .maxRetries(1)
+            .build();
+
+        FunctionCallingService service = new FunctionCallingService();
+
+        FunctionCallingAssistant assistant = AiServices.builder(FunctionCallingAssistant.class)
+            .chatLanguageModel(model)
+            .chatMemory(MessageWindowChatMemory.withMaxMessages(100))
+            .tools(service)
+            .build();
+
+        // when
+        String response = assistant.chat("What is the status of my payment transactions 001 and 002?");
+        System.out.println(response);
+
+        // then
+        assertThat(response).contains("001");
+        assertThat(response).contains("002");
+        assertThat(response).contains("pending");
+        assertThat(response).contains("approved");
+        assertThat(response).doesNotContain("003");
+        assertThat(response).doesNotContain("rejected");
+
+        // when
+        response = assistant.chat("What is the status of transactions 003?");
+        System.out.println(response);
+
+        // then
+        assertThat(response).doesNotContain("001");
+        assertThat(response).doesNotContain("002");
+        assertThat(response).doesNotContain("pending");
+        assertThat(response).doesNotContain("approved");
+        assertThat(response).contains("003");
+        assertThat(response).contains("rejected");
+    }
+
+    @Test
+    void should_use_google_search() {
+
+        // given
+        VertexAiGeminiChatModel modelWithSearch = VertexAiGeminiChatModel.builder()
+            .project("genai-java-demos")
+            .location("us-central1")
+            .modelName("gemini-1.5-flash-001")
+            .useGoogleSearch(true)
+            .build();
+
+        // when
+        String resp = modelWithSearch.generate(
+            "Why is the sky blue?");
+
+        // then
+        System.out.println("Google Search powered response = " + resp);
+        assertThat(resp).contains("scatter");
+    }
+
+    @Test
+    void should_support_json_response_mime_type() {
+
+        // given
+        VertexAiGeminiChatModel modelWithResponseMimeType = VertexAiGeminiChatModel.builder()
+            .project("genai-java-demos")
+            .location("us-central1")
+            .modelName("gemini-1.5-flash-001")
+            .responseMimeType("application/json")
+            .build();
+
+        // when
+        String response = modelWithResponseMimeType.generate("Run a dice roll");
+
+        // then
+        System.out.println("response = " + response);
+        assertThat(response).containsAnyOf("1", "2", "3", "4", "5", "6");
```

Comment:
This does not seem to prove that generated thing is a valid JSON, maybe copy-paste `OpenAiChatModelIT.should_generate_valid_json`?